app/navigation/Navigator.scala

```diff
@@ -213,6 +214,15 @@ class Navigator @Inject() (categorisationService: CategorisationService) {
       }
       .getOrElse(routes.JourneyRecoveryController.onPageLoad())
 
+  private def navigateFromRemoveNirmsPage(answers: UserAnswers): Call =
+    answers
+      .get(RemoveNirmsPage)
+      .map {
+        case true  => routes.CyaMaintainProfileController.onPageLoadNirms
+        case false => routes.ProfileController.onPageLoad()
```

Comment:
this worries me because when are you clearing the user answers?

maybe the clearing user answers should happen on page load of Profile controller in this case?

Reply:
It's already happens on page load of Profile controller